### ISSUE TITLE
Crash

### ISSUE THREAD

**Issue by GLoOoccK** [SmileYzn/LogApi#8]
@SmileYzn

I couldn't find a way to force the crash, it happens randomly, during map change

[debug.log](https://github.com/user-attachments/files/16910719/debug.log)

I'll try to figure out a way to force the crash

**Comment by SmileYzn** [comment]
Did you tested latest action build?

**Comment by GLoOoccK** [comment]
@SmileYzn

Yes, with the last action build

**Comment by SmileYzn** [comment]
What events are activated here?

**Comment by GLoOoccK** [comment]
```json
{
	"ServerActivate": 		true,
	"ServerDeactivate": 		true,
	"ServerAlertMessage": 		true,
	"ServerInfo": 			true,
	"ClientConnect":		true,
	"ClientPutInServer":		true,
	"ClientDisconnect":		true,
	"ClientKill": 			true,
	"ClientUserInfoChanged":	true,
	"ClientCommand": 		true,
	"ClientSay": 			true,
	"ClientMenuHandle":		true
}
```

**Comment by SmileYzn** [comment]
Rofl, Did you can try to disable some events?
I think that is related to serverDeactivate, if i remember serverFrame stops when server is changing level or something else.

**Comment by GLoOoccK** [comment]
Yes, I will gradually activate the events

**Comment by GLoOoccK** [comment]
@SmileYzn

The problem is being caused by the ClientCommand, mainly when changing maps, I tested it with the following configuration:

```json
{
	"ServerActivate": 		true,
	"ServerDeactivate": 		false,
	"ServerAlertMessage": 		true,
	"ServerInfo": 			true,
	"ClientConnect":		true,
	"ClientPutInServer":		true,
	"ClientDisconnect":		true,
	"ClientKill": 			true,
	"ClientUserInfoChanged":	true,
	"ClientCommand": 		true,
	"ClientSay": 			true,
	"ClientMenuHandle":		true
}
```
[debug.log](https://github.com/user-attachments/files/16922807/debug.log)

I no longer had any crashes with this configuration:

```json
{
	"ServerActivate": 		true,
	"ServerDeactivate": 		true,
	"ServerAlertMessage": 		true,
	"ServerInfo": 			true,
	"ClientConnect":		true,
	"ClientPutInServer":		true,
	"ClientDisconnect":		true,
	"ClientKill": 			true,
	"ClientUserInfoChanged":	true,
	"ClientCommand": 		false,
	"ClientSay": 			true,
	"ClientMenuHandle":		true
}
```

I found another problem, when i change maps it is not reading events.json again, it is only reading when the server starts for the first time

**Comment by SmileYzn** [comment]
Fixed events reload at changelevel here:

https://github.com/SmileYzn/LogApi/actions/runs/11001542230

Ps. Need to investigate crash issues

**Comment by GLoOoccK** [comment]
@SmileYzn

I will test the latest build, would it be possible to add keepalive support?

https://curl.se/libcurl/c/CURLOPT_TCP_KEEPALIVE.html

**Comment by SmileYzn** [comment]
Added in

https://github.com/SmileYzn/LogApi/actions/runs/11001951624

**Comment by GLoOoccK** [comment]
@SmileYzn

Server is crashing when starting on Ubuntu 22.04 LTS:

```
double free or corruption (fasttop)
Aborted
```

I tried with the last two versions

**Comment by SmileYzn** [comment]
Is the new curl option 🤦

**Comment by GLoOoccK** [comment]
@SmileYzn

This version works normally when starting the server:

https://github.com/SmileYzn/LogApi/actions/runs/10926424682

Can't it be something in the build.yml or Makefile?

**Comment by SmileYzn** [comment]
I updated log api today, still crashing?

**Comment by GLoOoccK** [comment]
@SmileYzn

Yes, all above [#35](https://github.com/SmileYzn/LogApi/commit/97220abc8ef5620993cc2c44b9b7a2d917721fcb) are crashing

**Comment by SmileYzn** [comment]
Removed CURLOPT_TCP_KEEPALIVE

If still crash, is not related with logApi

**Comment by SmileYzn** [comment]
Only logapi is running?

**Comment by SmileYzn** [comment]
Removed std::map container clean on start.
If still crash, the problem is not in logapi in really

**Comment by GLoOoccK** [comment]
When I leave only it, it works, when I turn on the AMX Mod X it starts crashing, the rest is off

```cpp
linux addons/logapi/dlls/logapi_mm.so
linux addons/amxmodx/dlls/amxmodx_mm_i386.so
```

**Comment by SmileYzn** [comment]
Updated CSSDK, may be can solve issues with amxmodx

**Comment by GLoOoccK** [comment]
@SmileYzn with the latest update, everything is working perfectly.

**Comment by SmileYzn** [comment]
🙏 

I guess that is because of server startup check for read json settings file.
Also update JSON hpp library 👍
